src/snowcli/cli/nativeapp/manager.py

```diff
@@ -97,38 +194,128 @@ def sync_deploy_root_with_stage(self, role: str, app_pkg: str):
+            )
+            sync_local_diff_with_stage(
+                role=role,
+                deploy_root_path=self.deploy_root,
+                diff_result=diff,
+                stage_path=self.stage_fqn,
+            )
+        return diff
+
+    def _apply_package_scripts(self) -> None:
+        """
+        Assuming the application package exists and we are using the correct role,
+        applies all package scripts in-order to the application package.
+        """
+        env = jinja2.Environment(
+            loader=jinja2.loaders.FileSystemLoader(self.project_root),
+            keep_trailing_newline=True,
+            undefined=jinja2.StrictUndefined,
         )
 
-    def app_run(self) -> SnowflakeCursor:
+        queued_queries = []
+        for relpath in self.package_scripts:
+            try:
+                template = env.get_template(relpath)
+                result = template.render(dict(package_name=self.package_name))
+                queued_queries.append(result)
+
+            except jinja2.TemplateNotFound as e:
+                raise MissingPackageScriptError(e.name)
+
+            except jinja2.TemplateSyntaxError as e:
+                raise InvalidPackageScriptError(e.name, e)
+
+            except jinja2.UndefinedError as e:
+                raise InvalidPackageScriptError(relpath, e)
+
+        # once we're sure all the templates expanded correctly, execute all of them
+        for queries in queued_queries:
+            self._execute_queries(queries)
 
-        project_name = clean_identifier(self.definition["name"])
-        app_pkg = self.definition.get("package", {}).get(
-            "name", default_app_package(project_name)
+    def _create_dev_app(self, diff: DiffResult) -> None:
+        """
+        (Re-)creates the application with our up-to-date stage.
+        """
+        app_role = (
+            self.definition.get("application", {}).get("role", None) or default_role()
+        )
+
+        with self.use_role(app_role):
+            if self.warehouse:
+                self._execute_query(f"use warehouse {self.warehouse}")
+
+            show_app_cursor = self._execute_query(
+                f"show applications like '{self.app_name}'", cursor_class=DictCursor
+            )
+
+            if show_app_cursor.rowcount != 0:
+                # There can only be one possible pre-existing app pkg with the same name
+                show_app_row: dict = show_app_cursor.fetchone()
+                if (
+                    show_app_row[COMMENT_COL] != SPECIAL_COMMENT
+                    or show_app_row[VERSION_COL] != LOOSE_FILES_MAGIC_VERSION
+                ):
+                    raise ApplicationAlreadyExistsError(self.app_name)
+
+                actual_owner = show_app_row[OWNER_COL]
+                if actual_owner != app_role:
+                    raise UnexpectedOwnerError(self.app_name, app_role, actual_owner)
```

Comment:
Will this ever happen with the current implementation?
We are using the context manager `with self.use_role(app_role):` and hence it will query `show applications` using this `app_role`. Even if the app exists but is not owned by `app_role`, it will not show up here. 

For `show` we should use `accountadmin` in order to actually verify if an object exists and its owner. 
(I need to make this change in my own PR too)

Reply:
I'm against using `accountadmin` because they may only have specialized roles and no access to that one. It's a good point re: ownership but if I'm not mistaken, any apps owned by roles where that role has been granted to the current "app role" will also show up here.

Just tested, and yes, you can see apps owned by other roles if those roles are granted to your role. So I think we should keep this logic so we're guaranteed the app is owned by the role you expect it to be.

Reply:
>  any apps owned by roles where that role has been granted to the current "app role" will also show up here.
I understand this, but I meant a case where owners are just two separate roles with no interaction with each other. Even if not accountadmin (where I take your point, you may not have access to it), but if an older `roleA` exists which owns an `appA`, and you're trying to query for an `application like 'appA'` from a `roleB`, and `roleA` is not granted to `roleB`, then it would not show up here in the results. But the creation would fail because `appA` still exists, regardless the ownership.

Reply:
Ohh I see your point now! Just tested, and in this case, we get this error:

> Application '<app_name>' already exists, but current role has no privileges on it. If this is unexpected and you cannot resolve this problem, contact your system administrator. ACCOUNTADMIN role may be required to manage the privileges on the object.

I think that's good enough for users to be able to action on, what do you think?